packages/app/tasks/get-pool-pending-rewards.js

```diff
@@ -0,0 +1,318 @@
+const fs = require('fs');
+const BN = require('bn.js');
+const { set, unset } = require('lodash');
+
+const { trace, info, error, warning, arg } = require('../utils/logger');
+const DB = require('../utils/db');
+
+const BATCH_SIZE = 500;
+
+const getPoolPendingRewardsTask = async (env, { poolToken }) => {
+    const eq = (address1, address2) => {
+        return address1.toLowerCase() === address2.toLowerCase();
+    };
+
+    const getPosition = async (id, blockNumber) => {
+        const position = await web3Provider.call(
+            contracts.LiquidityProtectionStore.methods.protectedLiquidity(id),
+            {},
+            blockNumber
+        );
+
+        return {
+            id,
+            provider: position[0],
+            poolToken: position[1],
+            reserveToken: position[2],
+            poolAmount: position[3],
+            reserveAmount: position[4],
+            reserveRateN: position[5],
+            reserveRateD: position[6],
+            timestamp: position[7]
+        };
+    };
+
+    const getProtectionLiquidityChanges = async (data, targetPoolToken, fromBlock, toBlock) => {
+        const poolData = {};
+
+        let eventCount = 0;
+        for (let i = fromBlock; i < toBlock; i += BATCH_SIZE) {
+            const endBlock = Math.min(i + BATCH_SIZE - 1, toBlock);
+
+            info(
+                'Querying all protection change events from',
+                arg('startBlock', i),
+                'to',
+                arg('endBlock', endBlock),
+                'in batches of',
+                arg('batchSize', BATCH_SIZE),
+                'blocks'
+            );
+
+            const events = await web3Provider.getPastEvents(contracts.LiquidityProtectionStore, 'allEvents', {
+                fromBlock: i,
+                toBlock: endBlock
+            });
+
+            for (const event of events) {
+                const { blockNumber, returnValues, transactionHash } = event;
+                const block = await web3Provider.getBlock(blockNumber);
```

Comment:
Maybe better to fetch these blocks outside of this `for` loop, using `Promise.all`.
If it's not too many of them (which will dump a huge chunk of data into the process memory), then it will potentially speed up the running time.
All of the above, under the assumption that you've designed this to communicate with a remote node and not with a local one.

Reply:
Yeah, it's not designed to work against a remote node - there is too much I/O and checks (especially in past/future scripts)